test/test_nestedtensor.py

```diff
@@ -578,6 +578,36 @@ def test_softmax(self, device, dtype):
         self.assertRaises(RuntimeError, lambda: torch.nn.functional.softmax(nt1, 0))
         self.assertRaises(IndexError, lambda: torch.nn.functional.softmax(nt1, 1))
 
+    @dtypes(torch.float, torch.float16, torch.double)
+    @torch.inference_mode()
+    def test_bmm(self, device, dtype):
+        # error case: not 3D tensors
+        nt0 = torch.nested_tensor([])
+        nt1 = torch.nested_tensor([torch.randn(2), torch.randn(3)])
+        nt2 = torch.nested_tensor([torch.randn((2, 4)), torch.randn((3, 4))])
+        self.assertRaises(RuntimeError, lambda: nt0.bmm(nt0))
```

Comment:
Bonus points for `assertRaisesRegex` so we can make sure we get the error message (and thus the error) that we expect. This prevents bugs more strongly, since `RuntimeError` is such a common Error.

Reply:
You can also orient yourself on the error messages that torch.bmm throws for regular torch.Tensors

Reply:
Rewritten the tests with `assertRaisesRegex`. 3D check error message is consistent.

The size check, however, I had to tweak a bit:
* regular torch.Tensor.bmm check the first 2 dimensions of batch2
* nested tensor has a well defined 1st dimension, but maybe not the 2nd dimension

See the `assertRaisesRegex` codes for the error messages I came up with